File: frontend/src/shared/models/execution-state-dto.ts
import type { RunStateEnum } from "@/shared/enums/backend/execution/run-state-enum";

/**
 * What the engine is doing right now. Read on mount, because the engine outlives the page and a
 * run started before you navigated away is still going.
 */
export interface ExecutionStateDto {
  state: RunStateEnum;
  isRunning: boolean;
  executionId: number;
  flowId: number;
}

File: backend/Core/Models/Dtos/ExecutionStateDto.cs
using Core.Enums;

namespace Core.Models.Dtos
{
    /// <summary>
    /// What the engine is doing right now. Read when the page mounts, so navigating away and back
    /// picks a run up where it is rather than showing an idle screen over a running flow.
    /// </summary>
    public class ExecutionStateDto
    {
        public RunStateEnum State { get; set; }
        public bool IsRunning { get; set; }

        public int ExecutionId { get; set; }
        public int FlowId { get; set; }
    }
}
